plugins/postgres/dbt/adapters/postgres/relation_configs/index.py

```diff
@@ -0,0 +1,174 @@
+    RelationConfigBase,
+    RelationResults,
+    RelationConfigValidationMixin,
+    RelationConfigValidationRule,
+    RelationConfigChangeAction,
+    RelationConfigChange,
+)
+
+
+# A `NodeConfig` instance can have multiple indexes, this is just one index
+# e.g. {"columns": ["column_a", "column_b"], "unique": True, "type": "hash"}
+Columns = List[str]
+ModelNodeEntry = Dict[str, Union[Columns, bool, str]]
+
+
+class PostgresIndexMethod(StrEnum):
+    btree = "btree"
+    hash = "hash"
+    gist = "gist"
+    spgist = "spgist"
+    gin = "gin"
+    brin = "brin"
+
+    @classmethod
+    def default(cls) -> "PostgresIndexMethod":
+        return cls.btree
+
+
+@dataclass(frozen=True, eq=True, unsafe_hash=True)
+class PostgresIndexConfig(RelationConfigBase, RelationConfigValidationMixin):
+    """
+    This config fallows the specs found here:
+    https://www.postgresql.org/docs/current/sql-createindex.html
+
+    The following parameters are configurable by dbt:
+    - name: the name of the index in the database, this isn't predictable since we apply a timestamp
+    - unique: checks for duplicate values when the index is created and on data updates
+    - method: the index method to be used
+    - column_names: the columns in the index
+
+    Applicable defaults for non-configurable parameters:
+    - concurrently: `False`
+    - nulls_distinct: `True`
+    """
+
+    name: Optional[str] = field(default=None, hash=False, compare=False)
+    column_names: Optional[FrozenSet[str]] = field(default_factory=set, hash=True)
+    unique: Optional[bool] = field(default=False, hash=True)
+    method: Optional[PostgresIndexMethod] = field(default=PostgresIndexMethod.btree, hash=True)
+
+    @property
+    def validation_rules(self) -> Set[RelationConfigValidationRule]:
+        return {
+            RelationConfigValidationRule(
+                validation_check=self.column_names is not None,
+                validation_error=DbtRuntimeError(
+                    "Indexes require at least one column, but none were provided"
+                ),
+            ),
+        }
+
+    @classmethod
+    def from_dict(cls, config_dict) -> "PostgresIndexConfig":
+        kwargs_dict = {
+            "name": config_dict.get("name"),
+            "method": config_dict.get("method"),
+            "unique": config_dict.get("unique"),
+            "column_names": frozenset(column for column in config_dict.get("column_names", {})),
+        }
+        index: "PostgresIndexConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return index
+
+    @classmethod
+    def parse_model_node(cls, model_node_entry: ModelNodeEntry) -> dict:
+        config_dict = {
+            "unique": model_node_entry.get("unique"),
+            "method": model_node_entry.get("type"),
+        }
+
+        if column_names := model_node_entry.get("columns", []):
+            # TODO: include the QuotePolicy instead of defaulting to lower()
```

Comment:
is this an outstanding TODO for this PR or is there a follow-up issue?

Reply:
Tomato tomahto. I will do this prior to release, but I think it can go into the beta as is. Ultimately that's your call.

Reply:
Cool -- as long as its tracked for prior to release 👍